terraform/aptos-node/aws/README.md

```diff
@@ -0,0 +1,181 @@
+
+3. Create Terraform file called `main.tf` in your working directory:
+  ```
+  $ cd ~/$WORKSPACE
+  $ touch main.tf
+  ```
+
+4. Modify `main.tf` file to configure Terraform, and create fullnode from Terraform module. Example content for `main.tf`:
+  ```
+  variable "aws_region" {
+    type    = string
+    default = "us-west-2"
+  }
+
+  terraform {
+    required_version = "~> 1.1.0"
+    backend "s3" {
+      bucket = "terraform.aptos-node"
+      key    = "state/aptos-node"
+      region = var.aws_region
+    }
+  }
+
+  provider "aws" {
+    region = var.aws_region
+  }
+
+  module "aptos-node" {
+    # download Terraform module from aptos-labs/aptos-core repo
+    source        = "github.com/aptos-labs/aptos-core.git//terraform/aptos-node/aws?ref=main"
+    region        = var.aws_region  # Specify the region
+    # zone_id     = "<Route53 zone id>"  # zone id for Route53 if you want to use DNS
+    era           = 1              # bump era number to wipe the chain
+    image_tag     = "testnet" # Specify the docker image tag to use
+    validator_name = "<Name of Your Validator>"
+  }
+  ```
+
+5. Initialise Terraform in the same directory of your `main.tf` file
+  ```
+  $ terraform init
+  ```
+This should download all the terraform dependencies for you, in the `.terraform` folder.
+
+6. Create a new Terraform workspace to isolate your environments:
+  ```
+  $ terraform workspace new $WORKSPACE
+  # This command will list all workspaces
+  $ terraform workspace list
+  ```
+
+7. Apply the configuration.
+  ```
+  $ terraform apply
+  ```
+  This might take a while to finish (~20 minutes), Terraform will create all the resources on your cloud account.
+
+8. Once terraform apply finishes, you can check if those resources are created:
+
+    - `aws eks update-kubeconfig --name aptos-$WORKSPACE` to configure access for your k8s cluster.
+    - `kubectl get pods` this should have haproxy, validator and fullnode. with validator and fullnode pod `pending` (require further action in later steps)
+    - `kubectl get svc` this should have `validator-lb` and `fullnode-lb`, with an external-IP you can share later for connectivity.
+
+9. Get your node IP info:
+
+    ```
+    $ export VALIDATOR_ADDRESS="$(kubectl get svc ${WORKSPACE}-aptos-node-validator-lb --output jsonpath='{.status.loadBalancer.ingress[0].hostname}')"
+
+    $ export FULLNODE_ADDRESS="$(kubectl get svc ${WORKSPACE}-aptos-node-fullnode-lb --output jsonpath='{.status.loadBalancer.ingress[0].hostname}')"
+    ```
+
+10. Generate key pairs (node owner key, consensus key and networking key) in your working directory.
+
+    ```
+    $ aptos genesis generate-keys --output-dir ~/$WORKSPACE
+    ```
+
+    This will create a `private-keys.yaml` file for you, Backup your key files somewhere safe, this is important for you to establish ownership of your node, and it will be used to claim your rewards later if eligible. Very important!!
```

Comment:
This will create a `private-keys.yaml` file for you. 

**IMPORTANT**: Backup your key files somewhere safe. This information is important for you to establish ownership of your node, and you will use it to claim your rewards later if eligible.